packages/passport-interface/src/api/requestDownloadAndDecryptStorage.ts

```diff
@@ -19,37 +34,102 @@ export async function requestDownloadAndDecryptStorage(
   zupassServerUrl: string,
   encryptionKey: string
 ): Promise<DownloadAndDecryptResult> {
+  const result = await requestDownloadAndDecryptUpdatedStorage(
+    zupassServerUrl,
+    encryptionKey,
+    undefined
+  );
+  if (!result.success) {
+    return { error: result.error, success: false };
+  }
+  if (!result.value.storage) {
+    console.error("unexpectedly missing e2ee data");
+    return {
+      error: {
+        name: ERROR_NAME_BAD_RESPONSE,
+        detailedMessage: "unexpectedly missing e2ee data"
+      },
+      success: false
+    };
+  }
```

Comment:
I can clarify why it's unexpected in the message.  My intent here is to check that the server is following the protocol properly.

For the unknown key, the server should return a 404 so we shouldn't reach this "success" case.

I don't think the behavior for that case is changing in this PR, but there is some risk of your account getting into a bad state if the first-time upload fails.  We upload in finishAccountCreation(), but I believe that by that point the account is "created" from the server's point of view.  However on future logins we have to download in order to check your password.  So if your account creation succeeds, but upload fails, you'll have an account you can't log into.  It's a brand-new account, though, so resetting it is probably a sufficient way to recover.

Reply:
I see. This edge case feels like the sort that may cause a 1/10000 bug.

Reply:
Agreed that's a possibility, though to clarify again I think that possibility is orthogonal to this PR.

I think the outcome of that 1/10000 bug isn't too dire.  So long as the user realizes that their account creation didn't finish, they can do a reset and be in a good state without losing anything.  I'm all for robustness, but probably 1/10000 bugs with minor impact can wait until we have 10000 users.

I feel like fixing it would require a more explicit relationship between E2EE storage creation and account creation.  We could make an explicit "finish account creation" step after the E2EE upload, and arrange for account creation to re-start if the user comes back after not finishing it.  Or it could be a single step if we made an explicit first-time-upload request to interact with both the E2EE storage database and the user database, in the same way that the password change flow does.  The latter would make more sense if the E2EE storage API were explicitly tied to a user and authenticated, allowing the server to be aware of whether a given user has created their E2EE storage or not.